packages/vscode-css-languageservice/src/parser/cssParser.ts

```diff
@@ -449,42 +497,102 @@ export class Parser {
 
 	public _parseDeclarations(parseDeclaration: () => nodes.Node | null): nodes.Declarations | null {
 		const node = this.create(nodes.Declarations);
-		if (!this.accept(TokenType.CurlyL)) {
-			return null;
+
+		if (this.syntax === "indented") {
+			if (!this.accept(TokenType.Indent)) {
+				return null;
+			}
+			while (this.accept(TokenType.Newline)) {
+				// accept empty statements
+			}
+		} else {
+			if (!this.accept(TokenType.CurlyL)) {
+				return null;
+			}
 		}
 
+		let initialDepth = this.scanner.stream.depth;
+
 		let decl = parseDeclaration();
 		while (node.addChild(decl)) {
-			if (this.peek(TokenType.CurlyR)) {
-				break;
-			}
-			if (this._needsSemicolonAfter(decl) && !this.accept(TokenType.SemiColon)) {
-				return this.finish(node, ParseError.SemiColonExpected, [TokenType.SemiColon, TokenType.CurlyR]);
-			}
-			// We accepted semicolon token. Link it to declaration.
-			if (decl && this.prevToken && this.prevToken.type === TokenType.SemiColon) {
-				(decl as nodes.Declaration).semicolonPosition = this.prevToken.offset;
+			if (this.syntax === "indented") {
+				if (this.peek(TokenType.Dedent)) {
+					break;
+				}
+				while (this.accept(TokenType.Newline)) {
+					// accept empty statements
+				}
+			} else {
+				if (this.peek(TokenType.CurlyR)) {
+					break;
+				}
+				if (this._needsSemicolonAfter(decl) && !this.accept(TokenType.SemiColon)) {
+					return this.finish(node, ParseError.SemiColonExpected, [TokenType.SemiColon, TokenType.CurlyR]); // It's this resync! It consumes stuff up until the syncToken. Need something similar for indented.
```

Comment:
```suggestion
					return this.finish(node, ParseError.SemiColonExpected, [TokenType.SemiColon, TokenType.CurlyR]);
```

I think I fixed this TODO 😅